app/models/service_template_ansible_playbook.rb

```diff
@@ -156,11 +156,17 @@ def update_catalog_item(options, auth_user = nil)
 
   def destroy
     auth_user = User.current_userid || 'system'
+    raise 'Attached retirement configurations do not allow the service template to be deleted.' if retirement_defined?
     resource_actions.where.not(:configuration_template_id => nil).each do |resource_action|
       job_template = resource_action.configuration_template
       ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript
         .delete_in_provider_queue(job_template.manager.id, { :manager_ref => job_template.manager_ref }, auth_user)
     end
     super
   end
+
+  def retirement_defined?
```

Comment:
Need a public method and a private one. The private one is used by `before_destroy` which may throw an error.